### PR TITLE
Add Python Binding for Psi4Real

### PR DESCRIPTION
**Pull request by AlexCarpenter46**
## Proposed changes

Add a python binding for Psi4Real

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/GeneralRelativity/Python/Psi4Real.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include "PointwiseFunctions/GeneralRelativity/Psi4Real.hpp"
+
+namespace py = pybind11;
+
+namespace GeneralRelativity::py_bindings {
+
+namespace {
+template <typename Frame>
+void bind_psi4real_impl(py::module& m) {  // NOLINT
+  m.def(
```
Comment (nilsvu) [inline src/PointwiseFunctions/GeneralRelativity/Python/Psi4Real.cpp]: you can put bindings of pointwise functions directly in the `Bindings.cpp`, that reduces the number of boilerplate / files considerably.

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: tests/Unit/PointwiseFunctions/GeneralRelativity/Python/Test_Bindings.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+from spectre.DataStructures.Tensor import tnsr, Frame
+
+import numpy as np
+import numpy.testing as npt
+import unittest
+
+
+class TestPsi4(unittest.TestCase):
+    def test_psi4(self):
+        spatial_ricci = tnsr.ii[DataVector, 3, Frame.Inertial](num_points=1,
```
Comment (nilsvu) [inline tests/Unit/PointwiseFunctions/GeneralRelativity/Python/Test_Bindings.py]: You can delete `Frame.Inertial` in all these

**Review by nilsvu — CHANGES_REQUESTED** [review]
Looks great! One small thing. Please squash everything into one commit.

File: tests/Unit/PointwiseFunctions/GeneralRelativity/Python/Test_Bindings.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from spectre.PointwiseFunctions.GeneralRelativity import psi4real
+from spectre.DataStructures import DataVector
+from spectre.DataStructures.Tensor import tnsr, Frame
+
+import numpy as np
+import numpy.testing as npt
+import unittest
+
+
+class TestPsi4(unittest.TestCase):
```
Comment (nilsvu) [inline tests/Unit/PointwiseFunctions/GeneralRelativity/Python/Test_Bindings.py:13]: name this `TestBindings`, but you can also do that in another PR.

**Review by nilsvu — APPROVED** [review]

File: tests/Unit/PointwiseFunctions/GeneralRelativity/Python/Test_Bindings.py (hunk 3 above)
Comment (AlexCarpenter46) [reply tests/Unit/PointwiseFunctions/GeneralRelativity/Python/Test_Bindings.py:13]: oh whoops, yeah I'll add that in the next pybinding PR for the gr functions

**Pull request merged**